### PR TITLE
Migrate tests to web test runner

### PR DESCRIPTION
**Pull request by quinnbudan**
This PR deletes `polymer test` and migrates all web component tests to `@open-wc/web-test-runner`. It also removes `polymer lint` and polymer CLI, so that we can make this package a module and replace all `mjs` files with `js` files (though I had issues with Jest and needed to make Jest's babel config `.cjs`)

### CODE REVIEW

File: test/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.html [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-
-			describe('d2l-activity-assignment-editor-detail', () => {
-				let element;
-
-				beforeEach(async() => {
-					element = fixture('basic');
-					await element.updateComplete;
-				});
-
-				it('should pass all axe tests', async() => {
```
Comment (TomMichielsD2L) [inline test/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.html:35]: Are we getting these tests for free elsewhere?

File: test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon-base.test.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	});
+
+	it('should display draft icon with tooltip with draft attribute', () => {
+		const draftIcon = draftEvaluation.shadowRoot.querySelector('d2l-icon');
+		const draftIconName = draftIcon.getAttribute('icon');
+		expect(draftIconName).to.equal('tier1:draft');
+
+		const draftIconToolTip = draftEvaluation.shadowRoot.querySelector('d2l-tooltip');
+		expect(draftIconToolTip.getAttribute('position')).to.equal('bottom');
+		expect(draftIconToolTip.getAttribute('offset')).to.equal('15');
```
Comment (StephaniePitman) [inline test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon-base.test.js:33]: nit: 15 can be seen as a magical number. I am on the fence about making it a constant though, because it is only used once.

File: test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.test.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { expect, fixture, html, waitUntil } from '@open-wc/testing';
+import sinon from 'sinon';
+import '../../components/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.js';
+import { setupFetchStub } from '../helpers.js';
+
+const activityDraftEvaluation = {
```
Comment (StephaniePitman) [inline test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.test.js:6]: nit: I dont remember if this is something we do in JS testing, but should we name these constants with capitals?

File: test/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.html (hunk 1 above)
Comment (quinnbudan) [reply test/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-editor-detail.html:35]: no, but I removed it because without an entity being fetched or loaded it doesn't render anything, and so I don't think the aXe tests actually do anything useful here. Some similar tests were kept around because it was as simple as moving them to .test.js and if you run them you'll see "Inapplicable Tests: 85", which means nothing is being tested

File: test/d2l-activity-editor/d2l-activity-attachments/d2l-activity-attachments-list.test.js [hunk 4]
```diff
@@ -38,7 +38,7 @@ describe('d2l-activity-attachment', function() {
 
 	describe('renders attachment list', () => {
 
-		it('passes accessibility test', async() => {
+		it.skip('passes accessibility test', async() => { // TODO: this test does not pass, skipped during migration to @open-wc/web-test-runner
```
Comment (TomMichielsD2L) [inline test/d2l-activity-editor/d2l-activity-attachments/d2l-activity-attachments-list.test.js:41]: :( Component needs work?

File: test/d2l-activity-name/d2l-activity-name.test.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
+import { expect, fixture, html, waitUntil } from '@open-wc/testing';
+import { setupFetchStub } from '../helpers.js';
+import sinon from 'sinon';
+import '../../components/d2l-activity-name/d2l-activity-name.js';
+
+const assignmentActivity = {
```
Comment (StephaniePitman) [inline test/d2l-activity-name/d2l-activity-name.test.js:6]: nit: constants' names as all caps, maybe

File: test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.test.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+		await waitUntil(() => evaluationIcon.entity, 'Entity failed to load');
+	}
+
+	before(async() => {
+		fetchStub = sinon.stub(window.d2lfetch, 'fetch');
+		setupFetchStub(fetchStub, hrefEntityMap);
+	});
+
+	it('should instantiate the element', async() => {
+		await loadComponent('activity-draft-evaluation', true);
```
Comment (TomMichielsD2L) [inline test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.test.js:48]: Doesn't load component only take one arg, what is the `true` for?

Comment (quinnbudan) [reply test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.test.js:48]: it's a hold over from when I copied this from somewhere else, it should be removed!

File: test/d2l-activity-list-item/d2l-activity-list-item.test.js [hunk 7]
```diff
@@ -169,48 +184,38 @@ describe('d2l-activity-list-item', () => {
...
+		it('Description is hidden at width 384', async() => {
+			await loadResponsiveComponent('/activity/1', '384px');
+			expect(component._showDescription).to.be.false;
+			const description = component.$$('#d2l-activity-list-item-description');
+			expect(description.hasAttribute('hidden')).to.be.true;
 		});
 
-		it('Description is not hidden at width 385', done => {
-			component = fixture('d2l-activity-list-item-responsive-385-fixture');
+		it('Description is not hidden at width 385', async() => {
```
Comment (TomMichielsD2L) [inline test/d2l-activity-list-item/d2l-activity-list-item.test.js:211]: The old test would wait for a render prior to checking for the change, is this new test a potential race condition?

Comment (TomMichielsD2L) [reply test/d2l-activity-list-item/d2l-activity-list-item.test.js:211]: Please disregard the previous comment, lets instead not check it twice.

File: test/d2l-quick-eval/behaviors/d2l-quick-eval-telemetry-behavior.test.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+		sandbox = sinon.createSandbox();
+	});
+
+	afterEach(function() {
+		sandbox.restore();
+	});
+
+	it('_logEvent nothing is returned with no event body', () => {
+		telemetryBehaviour.dataTelemetryEndpoint = 'https:??test.string.d2l';
+		expect(undefined).to.equal(telemetryBehaviour._logEvent());
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/behaviors/d2l-quick-eval-telemetry-behavior.test.js:27]: nit: This test and next have the expect reversed

File: test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.test.js (hunk 3 above)
Comment (dlockhart) [reply test/d2l-activity-evaluation-icon/d2l-activity-evaluation-icon.test.js:6]: I wouldn't worry too much about using uppercase unless it's a more traditional "top of the file" type of constant.

Like I'd use capitals for something like this:
```javascript
const NUM_RETRIES = 5; // then maybe this is used a few times throughout the file or project
const ACTIVITY_TYPES = {
  Quiz: 2,
  Assignment: 4,
  ...
}; // an enum-type constant I like using uppercase
```

But linters complain if you don't use `const` anytime a variable is set and not modified, so uppercase can become overkill in scenarios like this:
```javascript
const elem = document.querySelector('foo');
elem.focus();
```

File: test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js [hunk 9]
```diff
@@ -0,0 +1,349 @@
...
+				name: 'apply',
+				type: 'application/x-www-form-urlencoded',
+				method: 'GET',
+				fields : defaultParams
+			};
+
+			const customParams = { customParam1: 'custom', customParam2: 'custom2' };
+			sinon.stub(component, 'performSirenAction').callsFake(function(passedAction) {
+				const fields = passedAction.fields;
+				expect(4).to.equal(fields.length);
```
[the remark this answers is not in the document]
Comment (dlockhart) [reply test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js:150]: Agreed -- if this assertion fails the error message will be backwards (e.g. "expected 4 to equal 5"), which might be confusing.

File: test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js [hunk 10]
```diff
@@ -0,0 +1,490 @@
...
+
+		list._health = { isHealthy: true, errorMessage: '' };
+		expect(alert.hasAttribute('hidden')).to.be.true;
+		done();
+	});
+	it('when handling initial load failure, alert should pop up and alert should hide when alerts cleared', (done) => {
+		list._health = { isHealthy: false, errorMessage: 'failedToLoadData' };
+
+		const alert = list.shadowRoot.querySelector('.list-alert');
+		expect(false).to.equal(alert.hasAttribute('hidden'));
```
Comment (StephaniePitman) [inline test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js:355]: nit: I think this may not been something you did but I think we could also do expect().to.be.false

File: test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js (hunk 9 above)
[the remark this answers is not in the document]
Comment (quinnbudan) [reply test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js:150]: I'll swap them! Thank you both

File: test/d2l-quick-eval/behaviors/d2l-quick-eval-telemetry-behavior.test.js (hunk 8 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/behaviors/d2l-quick-eval-telemetry-behavior.test.js:27]: I should swap these!

File: test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js (hunk 10 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js:355]: Yeah that's definitely more idiomatic

File: test/d2l-quick-eval/d2l-quick-eval-activities.test.js [hunk 11]
```diff
@@ -0,0 +1,199 @@
...
+		act._loading = false;
+	});
+
+	it('instantiating the element works', function() {
+		expect(act.tagName.toLowerCase()).to.equal('d2l-quick-eval-activities');
+	});
+
+	it('attributes are set correctly', function() {
+		expect(act.href).to.equal('blah');
+		expect(act.token).to.equal('t');
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/d2l-quick-eval-activities.test.js:100]: I Guess if we are cleaning the tests up we might as well make this consistent with the other tests. Aka a token value of `token`

File: test/d2l-quick-eval/data/sorts.js [hunk 12]
```diff
@@ -0,0 +1,3 @@
+export default {
```
Comment (StephaniePitman) [inline test/d2l-quick-eval/data/sorts.js:1]: Did this one not exist before? Looks like most of these data js files started as a json, and this one does not seem to have one. Might of just scanned by it, but wanted to ask just in case

File: test/d2l-quick-eval/d2l-quick-eval-activities.test.js [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+	});
+
+	it('if there is no data, d2l-quick-eval-no-submissions-image is shown', () => {
+		expect(noSubmissionComponent.hasAttribute('hidden')).to.not.be.true;
+		expect(noCriteriaResultsComponent.hasAttribute('hidden')).to.be.true;
+		expect(list.hasAttribute('hidden')).to.be.true;
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/d2l-quick-eval-activities.test.js:106]: The original test was based on the style having a display value of none. It excepted that for all three components, would we not expect the NoSubmissionComponent to also be hidden for this test?

File: test/d2l-quick-eval/data/sorts.js (hunk 12 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/data/sorts.js:1]: ah yeah this one didn't exist! I created it because a test needed it once it was actually fetching and I lost the ability to put `D2L.Siren` into "test mode"

File: test/d2l-quick-eval/data/unassessedActivities.js [hunk 14]
```diff
@@ -0,0 +1,200 @@
...
+		{
+			'rel':['https://api.brightspace.com/rels/filters'],
+			'href':'data/filters.js'
+		},
+		{
+			'rel':['self'],
+			'href':'https://www.example.com/?pageSize=20&filter=96W3siU29ydCI6eyJJ&sort=Y3Rpb24iOjB9'
+		},
+		{
+			'rel': [
```
Comment (StephaniePitman) [inline test/d2l-quick-eval/data/unassessedActivities.js:30]: This part seems to be different from the json version of this. I am assuming this is just part of the work that was done, but just wanted to double check.

File: test/d2l-quick-eval/d2l-quick-eval-activities.test.js (hunk 11 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/d2l-quick-eval-activities.test.js:100]: I can definitely do that! I think in the end it's only important that it's consistent within the test though, after all there are ~50 or so test files and ~300 or so tests, some written very far apart by different people and I don't think they all need to be consistent (but definitely the quick-eval tests should be consistent with each other, and I think I changed some of them to `token` and not others, so it's 100% on me here)

File: test/d2l-quick-eval/data/unassessedActivities.js (hunk 14 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/data/unassessedActivities.js:30]: Yeah so back when fetching wasn't actually happening the component never tried to get the sorts entity, but with my changes it now does. The test was failing because the component couldn't render since it couldn't get this link, so I added it!

File: test/d2l-quick-eval/d2l-quick-eval-dismissed-activities.test.js [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+				dismissedDate: '2019-01-01T01:01:01.001Z',
+				type: 'discussion'
+			}
+		];
+	}
+
+	before(async() => {
+		act = await fixture(html`
+			<d2l-quick-eval-dismissed-activities href='blah' token='t'>
+			</d2l-quick-eval-dismissed-activities>
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/d2l-quick-eval-dismissed-activities.test.js:49]: nit: Another blah and t token

File: test/d2l-quick-eval/d2l-quick-eval-activities.test.js (hunk 13 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/d2l-quick-eval-activities.test.js:106]: I'm not sure where `getComputedStyle` was originally coming from, but I lost access to it, and `window.getComputedStyle` exists but was not working as expected (every style was always `''`). On debugging this test I noticed that the real difference was this `hidden` attribute being present or not. I figured that since the goal of the test was to make sure certain children were showing in certain conditions, this was a fine way to do it

File: test/d2l-quick-eval/d2l-quick-eval-logging.test.js [hunk 16]
```diff
@@ -0,0 +1,78 @@
...
+		`);
+		fetchStub = sinon.stub(window, 'fetch');
+	});
+
+	afterEach(() => {
+		fetchStub.reset();
+	});
+
+	it('sends POST request with expected options formatted in the correct way', () => {
+		fetchStub.callsFake((logger, options) => {
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/d2l-quick-eval-logging.test.js:20]: I think we still  need to call done at the end of this test? or is this synchronous?

Comment (TomMichielsD2L) [reply test/d2l-quick-eval/d2l-quick-eval-logging.test.js:20]: Same question applies for the whole file.

File: test/d2l-quick-eval/d2l-quick-eval-search-results-summary-container.test.js [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+				</d2l-quick-eval-search-results-summary-container>
+			`);
+		});
+
+		it('instantiating the element works', function() {
+			expect(component.tagName.toLowerCase()).to.equal('d2l-quick-eval-search-results-summary-container');
+		});
+
+		it('Default values are correct', function() {
+			expect(5).to.equal(component.searchResultsCount);
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/d2l-quick-eval-search-results-summary-container.test.js:20]: All these tests have the expectation and result flipped.

File: test/d2l-quick-eval/d2l-quick-eval-logging.test.js (hunk 16 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/d2l-quick-eval-logging.test.js:20]: it's syncrhonous, `_logError` just ends with calling `window.fetch` and so the test ends when this function ends

File: test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js [hunk 18]
```diff
@@ -0,0 +1,490 @@
...
+		const alert = list.shadowRoot.querySelector('.list-alert');
+		expect(alert.hasAttribute('hidden')).to.be.true;
+	});
+	it('showLoadingSkeleton is set to true, showLoadingSpinner is set to false before data is loaded, and loading-skeleton is present', () => {
+		const loadingskeleton = list.shadowRoot.querySelector('d2l-quick-eval-submissions-skeleton');
+		expect(loadingskeleton.hidden).to.be.false;
+		expect(list.showLoadingSpinner).to.be.false;
+		expect(list.showLoadingSkeleton).to.be.true;
+	});
+	it.skip('_fullListLoading and _loading is set to false after data is loaded and the loading skeleton is hidden', (done) => {
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js:166]: nit: Looks like these tests have been skipped for 3 years now. Id be tempted to remove them, that might make this file closer to a reasonable size

File: test/d2l-quick-eval/d2l-quick-eval-search-results-summary-container.test.js (hunk 17 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/d2l-quick-eval-search-results-summary-container.test.js:20]: I shall flip them

File: test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js (hunk 18 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js:166]: On the one hand I'm hesitant to delete tests that our team doesn't own, but 3 years is a pretty long time, so I'm happy to remove them, especially if a quick-eval owner chimes in and agrees 😄

File: test/d2l-quick-eval/d2l-quick-eval-view-toggle.test.js [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+		it('should fire d2l-quick-eval-view-toggle-changed event when activities is selected and submissions is currently active', function(done) {
+			toggle.currentSelected = 'submissions';
+			toggle.addEventListener('d2l-quick-eval-view-toggle-changed', function() {
+				expect('activities').to.equal(toggle.currentSelected);
+				done();
+			});
+
+			toggle._selectActivities();
+		});
+		it('should not fire d2l-quick-eval-view-toggle-changed event when activities is selected and activities is currently active', function() {
```
Comment (TomMichielsD2L) [inline test/d2l-quick-eval/d2l-quick-eval-view-toggle.test.js:41]: not your fault, but I doubt this test does what it intends to do. We never wait any amount of time to validate the event is not fired. Im tempted to remove this test because it is a lie

Comment (quinnbudan) [reply test/d2l-quick-eval/d2l-quick-eval-view-toggle.test.js:41]: yeah not only does this test do nothing, but it should have failed due to `assert`, and also the thing that this test wants to test (based on the description) doesn't seem to be possible (testing that setting an attribute results in nothing happening)

File: jest.babel.config.cjs [hunk 20]
```diff
@@ -9,5 +9,5 @@ module.exports = require('babel-jest').createTransformer({
 				}
 			}
 		]
-	]
+	],
```
Comment (dlockhart) [inline jest.babel.config.cjs:12]: Nit: unnecessary diff, would make it 210 files instead of 211! ;)

File: package.json [hunk 21]
```diff
@@ -1,25 +1,23 @@
...
   "description": "Contains Hypermedia components that display informations about activites",
+  "type": "module",
   "repository": {
     "type": "git",
     "url": "https://github.com/BrightspaceHypermediaComponents/activities.git"
   },
   "name": "d2l-activities",
   "scripts": {
-    "lint": "npm run lint:style && npm run lint:wc && npm run lint:js && npm run lint:langs",
+    "lint": "npm run lint:style && npm run lint:js",
```
Comment (dlockhart) [inline package.json:10]: Do you not still want the `npm run lint:langs` tests?

File: package.json [hunk 22]
```diff
@@ -51,8 +49,8 @@
     "jest": "^26.0.1",
     "lie": "^3.3.0",
     "messageformat-validator": "^2.0.4",
-    "polymer-cli": "^1.9.2",
```
Comment (dlockhart) [inline package.json:54]: 🎉

File: package.json [hunk 23]
```diff
@@ -51,8 +49,8 @@
...
     "sinon": "^9.0.1",
+    "siren-parser": "^8.5.1",
     "stylelint": "^13.6.1",
     "url-polyfill": "^1.1.3",
     "wct-browser-legacy": "^1.0.1",
```
Comment (dlockhart) [inline package.json:56]: I think `wct-browser-legacy`, `wct-local` and `wct-mocha` can go away now too?

Also if you want to clean up more in a separate PR: `whatwg-fetch` and `webcomponentsjs` can go away since all the browsers we care about support fetch/WCs and we don't need the polyfills.

File: test/d2l-activity-admin-list/d2l-activity-admin-list.test.js [hunk 24]
```diff
@@ -1,11 +1,15 @@
 import { runAxe } from '@brightspace-ui/core/tools/a11y-test-helper.js';
```
Comment (dlockhart) [inline test/d2l-activity-admin-list/d2l-activity-admin-list.test.js:1]: I'd suggest using `@open-wc`'s built-in aXe stuff:

```javascript
import { fixture, expect, html } from '@open-wc/testing';

it('should pass all axe tests', async () => {
  const el = await fixture(element);
  await expect(el).to.be.accessible();
});
```

File: test/d2l-activity-editor/d2l-activity-attachments/d2l-activity-attachments-editor.html [hunk 25]
```diff
@@ -1,16 +0,0 @@
...
-		<meta name="viewport" content="width=device-width, minimum-scale=1, initial-scale=1, user-scalable=yes">
-		<title>d2l-activity-editor-attachments-picker tests</title>
-		<script src="/node_modules/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
-		<script src="/node_modules/mocha/mocha.js"></script>
-		<script src="/node_modules/chai/chai.js"></script>
-		<script src="/node_modules/wct-mocha/wct-mocha.js"></script>
-		<script src="/node_modules/axe-core/axe.min.js"></script>
-	</head>
-	<body>
-		<script type="module" src="./d2l-activity-attachments-editor.js"></script>
```
Comment (dlockhart) [inline test/d2l-activity-editor/d2l-activity-attachments/d2l-activity-attachments-editor.html:14]: So awesome that a lot of these were just a matter of deleting the HTML file and renaming the JS file!

File: test/d2l-quick-eval/compatability/ie11shims.test.js [hunk 26]
```diff
@@ -1,11 +1,12 @@
+import { expect } from '@open-wc/testing';
 import {
 	DictToQueryString,
 	StringEndsWith,
 	GetQueryStringParams,
 	GetQueryStringParam
 } from '../../../components/d2l-quick-eval/compatability/ie11shims.js';
```
Comment (dlockhart) [inline test/d2l-quick-eval/compatability/ie11shims.test.js:7]: IE11 is no longer supported... if that helps haha.

**Review by dlockhart — DISMISSED** [review]
This looks really great to me, really awesome job grinding through what I can only imagine was... a grind to get these ported over. It'll be so nice though to just have 1 testing framework in here and more importantly for Polymer CLI to be out of this repo. Kudos! 💯

File: test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js (hunk 18 above)
Comment (dlockhart) [reply test/d2l-quick-eval/d2l-quick-eval-submissions-table.test.js:166]: I'm not a quick-eval owner, but I'd be on bored with just removing them at this point as well. All original value in them has been lost.

File: package.json (hunk 21 above)
Comment (quinnbudan) [reply package.json:10]: I do! They're broken on master, and I merged master into my branch since I had a conflict, so I have temporarily removed it just so I could still verify CI was working and I plan on adding it back before merging. I told my team about this but not you, sorry!

File: package.json (hunk 23 above)
Comment (quinnbudan) [reply package.json:56]: Good point! I'll remove those now! And I'd definitely like to get to those later, deleting stuff feels good 😆

File: test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js [hunk 27]
```diff
@@ -0,0 +1,349 @@
...
+				method: 'GET',
+				fields : [
+					{
+						type: 'hidden',
+						name : 'existingField',
+						value: 'existingValue'
+					}]
+			};
+
+			const stub = sinon.stub(component, 'performSirenAction').callsFake(function(passedAction) {
```
Comment (StephaniePitman) [inline test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js:48]: nit: maybe could be a helper function

File: test/d2l-activity-admin-list/d2l-activity-admin-list.test.js (hunk 24 above)
Comment (quinnbudan) [reply test/d2l-activity-admin-list/d2l-activity-admin-list.test.js:1]: I'll find all instances of `runAxe` and replace them

File: test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js (hunk 27 above)
Comment (StephaniePitman) [reply test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js:48]: But I understand this isnt your test, so I am not stuck on this idea

File: test/d2l-quick-eval/compatability/ie11shims.test.js (hunk 26 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/compatability/ie11shims.test.js:7]: Yeah I was wondering about this while I was porting this test... I've decided not to delete them because quick eval still uses all the shims, and I didn't want to risk removing all that stuff. And since I was keeping that I figured keeping the test makes sense too

**Review by StephaniePitman — DISMISSED** [review]

File: package.json (hunk 23 above)
Comment (dlockhart) [reply package.json:56]: It's amazing yeah! For the fetch/WC polyfills, just search for them being included in the demo & test pages as scripts and delete those as well and you're golden.

File: test/d2l-quick-eval/compatability/ie11shims.test.js (hunk 26 above)
Comment (dlockhart) [reply test/d2l-quick-eval/compatability/ie11shims.test.js:7]: Yeah totally fair, better to keep this change as isolated as possible. But definitely can go on the list of old dead code to remove later.

File: test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js (hunk 27 above)
Comment (quinnbudan) [reply test/d2l-quick-eval/behaviors/d2l-siren-helper-behavior.test.js]: I pulled it out! Nice catch

File: package.json (hunk 23 above)
Comment (quinnbudan) [reply package.json]: I ended up also removing `lie` and `url-polyfill`, since those only seemed to be used in the demo pages as well under comments that said "For IE11". After a quick smoke test the demo pages seem to still be working!

File: test/d2l-activity-editor/d2l-activity-attachments/d2l-activity-attachments-list.test.js (hunk 4 above)
Comment (quinnbudan) [reply test/d2l-activity-editor/d2l-activity-attachments/d2l-activity-attachments-list.test.js:41]: it might have passed before because it may have been using an outdated way of checking accessibility? It was loading `node_modules/axe-core/axe.min.js` in the old version

**Review by TomMichielsD2L — APPROVED** [review]
